### ISSUE TITLE
Moving over to a Trilium instance for documentation?

### ISSUE THREAD

**Issue by eliandoran** [TriliumNext/Docs#117]
@meichthys , @Alumniminium , now that there is PikaPods, I think we can consider moving the documentation over to a shared instance of Trilium with only a few people having edit access. The public will be able to access it via the share API; what's your take on it?

I think there are a few blockers that we might want to address before:

- The default share theme is not very user-friendly. I think @zerebos wanted to contribute upstream to Trilium Notes an improved share theme but I'm not sure if he had the time to do it.
- The URLs are not very user-friendly because the note IDs are randomly generated. It might be interesting to allow the user to define custom friendly URLs for shared notes, but it would involve a bit of development.

cc. @maphew

**Comment by maphew** [comment]
**theme**: +1. and work on making theming easier to change and apply will be welcome for all users anyway.

**Friendly urls**: perhaps a script with button/keystroke for "turn title into share-alias". One better: it's actually the default on save, but only acts if `#shareAlias` is missing/empty. In case of duplicate append something.

**url generally**:  How does trilium.rocks bare domain serve shared pages? There's no `share/` prefix. And then how does one get to the edit interface? (I'd like to do the same at my domain, but the bare url is a login prompt). Also see https://github.com/orgs/TriliumNext/discussions/494

**Comment by maphew** [comment]
re: default on-save: that idea would add needless performance hit. Better to tie it to "when shared" slider which is a one time or at least user initiated thing.

**Comment by maphew** [comment]
I agree that none of the items listed so far are blockers.

On Pika hosting: since I already have a dialogue with I'll ask them about it.

**Comment by m3nu** [comment]
You guys are welcome to run as many pods you need for free on our platform. For demo, docs, etc.

**Comment by eliandoran** [comment]
@maphew ,

> Friendly urls: perhaps a script with button/keystroke for "turn title into share-alias". One better: it's actually the default on save, but only acts if #shareAlias is missing/empty. In case of duplicate append something.

An important aspect is that changing the friendly URL cannot be done automatically, because the user might not be aware that changing the title will also affect the URL. Generally once a page is published it's a good idea to keep the URL the same in order not to break any hard links (inside the application, or bookmarked by viewers of the site).

> url generally: How does trilium.rocks bare domain serve shared pages? There's no share/ prefix. And then how does one get to the edit interface? (I'd like to do the same at my domain, but the bare url is a login prompt). Also see https://github.com/orgs/TriliumNext/discussions/494

Most likely it's simply a reverse proxy configuration. Whenever the domain is accessed, it proxies the URLs to a local instance with the `share/` prefix. Of course, this is only an assumption, but it's most likely.

**Comment by eliandoran** [comment]
> You guys are welcome to run as many pods you need for free on our platform. For demo, docs, etc.

@m3nu , your help is much appreciated.

I will create an account and create a pod to manage the documentation as a proof of concept. We'll later see with the rest of the team what they think about it.

**Comment by maphew** [comment]
> An important aspect is that changing the friendly URL cannot be done automatically, because the user might not be aware that changing the title will also affect the URL.

Agreed, the auto apply of shareAlias only happens once, and after that is only changed by a deliberate user act and not by the system.

**Comment by eliandoran** [comment]
@Alumniminium , @meichthys , please reach out to my via an email to contact@eliandoran.me in order to share credentials.

**Comment by perfectra1n** [comment]
I worked with Zerebos to set up some documentation on how to set up the share theme [here](https://trilium.rocks/xecUJ8eL3pvM), and I've uploaded the "full" set of Notes required to use the theme [here](https://github.com/perfectra1n/trilium.rocks/releases/tag/1.0.0) along with some instructions. It works super well, I love the theme :)

**Comment by Alumniminium** [comment]
Sent the mail, sounds perfect. I'm all for it.

**Comment by eliandoran** [comment]
@perfectra1n , wonderful job.

**Comment by eliandoran** [comment]
> Sent the mail, sounds perfect. I'm all for it.

@Alumniminium , replied. Let me know if you don't receive it, since I had some server logs about the message being filtered.

**Comment by Alumniminium** [comment]
> > Sent the mail, sounds perfect. I'm all for it.
> 
> @Alumniminium , replied. Let me know if you don't receive it, since I had some server logs about the message being filtered.

all good :) thanks!

**Comment by maphew** [comment]
What is the status on the effort to move the docs repo into a hosted Trilium instance sit? 
Is there something I can do to help?

**Comment by maphew** [comment]
Ah, thank you @meichthys for reminding me https://github.com/TriliumNext/Notes/issues/658 was still open, yes it's closed as of PR https://github.com/TriliumNext/Notes/pull/1207.

Near the end @eliandoran suggested:

>instead of simply redirecting to /share, wouldn't it make more sense to also rewrite all addresses such as /share/STi3RCMhUvG6 to /STi3RCMhUvG6?

...and I argued (lightly) for keeping `/share` for making it easy for the author to tell what's private and public. In the case of Trilium Docs though I think the argument flips the other way. 

My first attempt at this, https://github.com/TriliumNext/Notes/pull/667, included an option to make the redirect target url configurable. I abandoned because it made things too complicated. Now that I know more I'll take another run at it.

**Comment by maphew** [comment]
> My first attempt at this, [TriliumNext/Notes#667](https://github.com/TriliumNext/Notes/pull/667), included an option to make the redirect target url configurable. I abandoned because it made things too complicated. Now that I know more I'll take another run at it.

https://github.com/TriliumNext/Notes/pull/1288 - some progress, but not working yet.
